mlflow/system_metrics/metrics/gpu_monitor.py

```diff
@@ -38,21 +38,24 @@ def __init__(self):
     def collect_metrics(self):
         # Get GPU metrics.
         for i, handle in enumerate(self.gpu_handles):
-            memory = pynvml.nvmlDeviceGetMemoryInfo(handle)
-            self._metrics[f"gpu_{i}_memory_usage_percentage"].append(
-                round(memory.used / memory.total * 100, 1)
-            )
-            self._metrics[f"gpu_{i}_memory_usage_megabytes"].append(memory.used / 1e6)
-
-            device_utilization = pynvml.nvmlDeviceGetUtilizationRates(handle)
-            self._metrics[f"gpu_{i}_utilization_percentage"].append(device_utilization.gpu)
-
-            power_milliwatts = pynvml.nvmlDeviceGetPowerUsage(handle)
-            power_capacity_milliwatts = pynvml.nvmlDeviceGetEnforcedPowerLimit(handle)
-            self._metrics[f"gpu_{i}_power_usage_watts"].append(power_milliwatts / 1000)
-            self._metrics[f"gpu_{i}_power_usage_percentage"].append(
-                (power_milliwatts / power_capacity_milliwatts) * 100
-            )
+            try:
+                memory = pynvml.nvmlDeviceGetMemoryInfo(handle)
+                self._metrics[f"gpu_{i}_memory_usage_percentage"].append(
+                    round(memory.used / memory.total * 100, 1)
+                )
+                self._metrics[f"gpu_{i}_memory_usage_megabytes"].append(memory.used / 1e6)
+
+                device_utilization = pynvml.nvmlDeviceGetUtilizationRates(handle)
+                self._metrics[f"gpu_{i}_utilization_percentage"].append(device_utilization.gpu)
+
+                power_milliwatts = pynvml.nvmlDeviceGetPowerUsage(handle)
+                power_capacity_milliwatts = pynvml.nvmlDeviceGetEnforcedPowerLimit(handle)
+                self._metrics[f"gpu_{i}_power_usage_watts"].append(power_milliwatts / 1000)
+                self._metrics[f"gpu_{i}_power_usage_percentage"].append(
+                    (power_milliwatts / power_capacity_milliwatts) * 100
+                )
+            except pynvml.nvml.NVMLError as e:
+                _logger.warning(f"Encountered error {e} when trying to collect GPU metrics.")
```

Comment:
```suggestion
                _logger.warning(f"Encountered error {e} when trying to collect GPU metrics for {handle}.")
```

not sure if this is helpful.

Reply:
yea mostly users have no action items on the exceptions, this is a safeguard on flaky NVIDIA stack.